homeassistant/components/fritz/__init__.py

```diff
@@ -1 +1,79 @@
-"""The fritz component."""
+"""Support for AVM Fritz!Box functions."""
+import asyncio
+import logging
+
+from fritzconnection.core.exceptions import FritzConnectionException, FritzSecurityError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.typing import ConfigType
+
+from .common import FritzBoxTools
+from .const import DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up fritzboxtools from config entry."""
+    _LOGGER.debug("Setting up FRITZ!Box Tools component")
+    fritz_tools = FritzBoxTools(
+        hass=hass,
+        host=entry.data[CONF_HOST],
+        port=entry.data[CONF_PORT],
+        username=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+    )
+
+    try:
+        await fritz_tools.async_setup()
+        await fritz_tools.async_start()
+    except FritzSecurityError as ex:
+        raise ConfigEntryAuthFailed from ex
+    except FritzConnectionException as ex:
+        raise ConfigEntryNotReady from ex
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = fritz_tools
+
+    @callback
+    def _async_unload(event):
+        fritz_tools.async_unload()
+
+    entry.async_on_unload(
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _async_unload)
+    )
+    # Load the other platforms like switch
+    for domain in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, domain)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigType) -> bool:
+    """Unload FRITZ!Box Tools config entry."""
+    fritzbox: FritzBoxTools = hass.data[DOMAIN][entry.entry_id]
+    fritzbox.async_unload()
```

Comment:
I think we miss an await here?!

Reply:
Thanks @AaronDavidSchneider 

@chemelli74 

The function is marked as a `@callback` and `async`. The `async` needs to get dropped since nothing is being awaited in `async_unload`

Reply:
There is also a bug introduced by the PR to fix the pylint mid-air collision in (see https://github.com/home-assistant/core/pull/49655#discussion_r619837558) that should be fixed

Reply:
PR created: #49669